.github/workflows/release.yml

```diff
@@ -254,95 +254,6 @@ jobs:
             --title ${{ github.ref_name }} \
             --generate-notes ${{ env.PRERELEASE }}
 
-  create-go-sdk-tag:
-    name: create tag sdk/go/v*
-    runs-on: ubuntu-latest
-    needs: build-and-sign
-    if: startsWith(github.ref, 'refs/tags/v')
-    steps:
-      - uses: actions/checkout@v3
-
-      - name: Set the tag to sdk/go/v*
-        shell: bash
-        run: echo "GO_SDK_TAG=sdk/go/${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
-
-      - name: Tag sdk/go/v* and push it
-        shell: bash
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          git tag ${{ env.GO_SDK_TAG }}
-          git push origin ${{ env.GO_SDK_TAG }}
-
-  create-template-sdk-update-pr:
-    name: Create PR with template SDK updates
-    runs-on: ubuntu-latest
-    needs: create-go-sdk-tag
-    if: startsWith(github.ref, 'refs/tags/v')
-    steps:
-      - uses: actions/checkout@v3
-
-      - name: Set the spin tag
-        shell: bash
-        run: |
-          echo "SPIN_TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
-
-      - name: Set the PR base branch
-        shell: bash
-        run: |
-          IFS=. read -r major minor patch <<< "${{ env.SPIN_TAG }}"
-          echo "RELEASE_BRANCH=$major.$minor" >> $GITHUB_ENV
-
-      - name: Change sdk version
-        shell: bash
-        run: |
-          cd templates
-          SDK_VERSION=${{ env.SPIN_TAG }} make
-
-      - name: Import GPG key
-        uses: crazy-max/ghaction-import-gpg@v5
-        with:
-          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
-          passphrase: ${{ secrets.PASSPHRASE }}
-          git_user_signingkey: true
-          git_commit_gpgsign: true
-
-      - name: Create Pull Request
-        uses: peter-evans/create-pull-request@v4
-        with:
-          commit-message: "feat(templates): update sdk to ${{ env.SPIN_TAG }}"
-          title: "feat(templates): update sdk to ${{ env.SPIN_TAG }}"
-          body: Update the SDK version used by the templates
-          branch: update-sdk-${{ env.SPIN_TAG }}
-          base: ${{ env.RELEASE_BRANCH }}
-          delete-branch: true
-          committer: fermybot <103076628+fermybot@users.noreply.github.com>
-          author: fermybot <103076628+fermybot@users.noreply.github.com>
-          signoff: true
-
-  # This will run when the PR above is approved and merged into main via a merge commit
-  push-templates-tag:
```

Comment:
Double-checking with @itowlson on the subject of templates:

For go, Assuming the v3 version of the sdk in its new home is a ways out and there may be a handful of subsequent Spin releases (eg 2.3, 2.4, etc) before it is ready, do we need to do anything ensure templates function as expected?  For as long as the go templates are in this repo and no new templates tags are pushed, will future spin clients just get the last, most recent templates (2.2) for this interim period?

The rust templates meanwhile should be ready to move to their new sdk home, right?  Would we update the `spin templates` cli commands to add the new sdk repo as a default to fetch from?

(Or maybe the go templates are also ready to migrate to their new sdk home as well -- especially if spin clients just fetch the 2.2-tagged templates near-term.)

Reply:
If template tags are not updated, then post-2.2 Spin releases will fall forward to the templates on `main`.

Moving the templates has implications for the installer, docs, and `spin new`.  I am wary of moving those without careful coordination.

Reply:
OK, let's assume we won't move the templates for the foreseeable future.  In that case, I guess our release process could look something like this:

- SDKs are released on their own schedule; for example, let's say we release `spin-rust-sdk` 3.0.0 tomorrow
- Templates remain tied to Spin releases, so they are _not_ updated until the next Spin release; e.g. Spin 2.3 would be our first opportunity to update the Rust templates to use `spin-rust-sdk` 3.0.0 (although I guess we could update them in `main` sooner than that)
    - i.e. when we make a Spin release, we tag the templates as usual, and those templates will point to whatever the latest stable versions of the SDKs happen to be

Does that sound right?  (No strong opinions here -- just trying to form a plan of some kind)

Reply:
We have to tag templates _at minimum_ when we do a (non-patch) Spin release.

We can _optionally_ update the templates and move the tags when we do a SDK release, even if no Spin release is planned.  But that's not part of the Spin release process and can be ad-hocked in if we ever feel the urge to actually do it.

Reply:
> i.e. when we make a Spin release, we tag the templates as usual, and those templates will point to whatever the latest stable versions of the SDKs happen to be

That sounds reasonable to me.  So, we'd re-introduce the corresponding automation around pushing the templates tag (eg `spin/templates/v2.3`) on a Spin release for the near-term?

Reply:
Er, maybe just keep `push-templates-tag` and remove `create-template-sdk-update-pr`, since we _don't_ want to update SDK versions automatically.

Reply:
e.g.:
```yaml
  push-templates-tag:
    runs-on: ubuntu-latest
    needs: build-and-sign
    if: startsWith(github.ref, 'refs/tags/v')
    steps:
      - uses: actions/checkout@v3

      - name: Set the tag to spin/templates/v*
        shell: bash
        run: |
          spin_tag=$(echo "${{ github.ref }}" | grep -Eo "refs/tags/v[0-9.]+")
          IFS=. read -r major minor patch <<< "${spin_tag}"
          echo "TEMPLATE_TAG=spin/templates/$major.$minor" >> $GITHUB_ENV

      - name: Tag spin/templates/v* and push it
        shell: bash
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          git tag ${{ env.TEMPLATE_TAG }} -f
          git push origin ${{ env.TEMPLATE_TAG }} -f
```

Reply:
I believe I've addressed this in https://github.com/fermyon/spin/pull/2289/commits/65ed372163af47bd556473f30cb420506a3bd603, but not sure how to test it.  Maybe push a release tag to my fork?

Reply:
Looks like it worked: https://github.com/dicej/spin/actions/runs/7995470091/job/21835889372

Reply:
Yes, that should be it... thanks for testing!